include/zephyr/drivers/gnss.h

```diff
@@ -0,0 +1,497 @@
+/*
+ * Copyright (c) 2023 Trackunit Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file gnss.h
+ * @brief Public GNSS API.
+ */
+
+#ifndef ZEPHYR_INCLUDE_DRIVERS_GNSS_H_
+#define ZEPHYR_INCLUDE_DRIVERS_GNSS_H_
+
+/**
+ * @brief GNSS Interface
+ * @defgroup gnss_interface GNSS Interface
+ * @ingroup io_interfaces
+ * @{
+ */
+
+#include <zephyr/types.h>
+#include <zephyr/device.h>
+#include <zephyr/data/navigation.h>
+#include <errno.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** GNSS PPS modes */
+enum gnss_pps_mode {
+	/** PPS output disabled */
+	GNSS_PPS_MODE_DISABLED = 0,
+	/** PPS output always enabled */
+	GNSS_PPS_MODE_ENABLED = 1,
+	/** PPS output enabled from first lock */
+	GNSS_PPS_MODE_ENABLED_AFTER_LOCK = 2,
+	/** PPS output enabled while locked */
+	GNSS_PPS_MODE_ENABLED_WHILE_LOCKED = 3
+};
+
+/** API for setting fix rate */
+typedef int (*gnss_set_fix_rate_t)(const struct device *dev, uint32_t fix_interval_ms);
+
+/** API for getting fix rate */
+typedef int (*gnss_get_fix_rate_t)(const struct device *dev, uint32_t *fix_interval_ms);
+
+/**
+ * @brief GNSS periodic tracking configuration
+ *
+ * @note Setting either active_time or inactive_time to 0 will disable periodic
+ * function.
+ */
+struct gnss_periodic_config {
+	/** The time the GNSS will spend in the active state in ms */
+	uint32_t active_time_ms;
+	/** The time the GNSS will spend in the inactive state in ms */
+	uint32_t inactive_time_ms;
+};
+
+/** API for setting periodic tracking configuration */
+typedef int (*gnss_set_periodic_config_t)(const struct device *dev,
+					  const struct gnss_periodic_config *periodic_config);
+
+/** API for setting periodic tracking configuration */
+typedef int (*gnss_get_periodic_config_t)(const struct device *dev,
+					  struct gnss_periodic_config *periodic_config);
+
+/** GNSS navigation modes */
+enum gnss_navigation_mode {
+	/** Dynamics have no impact on tracking */
+	GNSS_NAVIGATION_MODE_ZERO_DYNAMICS = 0,
+	/** Low dynamics have higher impact on tracking */
+	GNSS_NAVIGATION_MODE_LOW_DYNAMICS = 1,
+	/** Low and high dynamics have equal impact on tracking */
+	GNSS_NAVIGATION_MODE_BALANCED_DYNAMICS = 2,
+	/** High dynamics have higher impact on tracking */
+	GNSS_NAVIGATION_MODE_HIGH_DYNAMICS = 3
+};
+
+/** API for setting navigation mode */
+typedef int (*gnss_set_navigation_mode_t)(const struct device *dev,
+					  enum gnss_navigation_mode mode);
+
+/** API for getting navigation mode */
+typedef int (*gnss_get_navigation_mode_t)(const struct device *dev,
+					  enum gnss_navigation_mode *mode);
+
+/**
+ * @name GNSS Systems
+ *
+ * @brief Systems contained in gnss_systems_t
+ *
+ * @{
+ */
+enum gnss_system {
+	GNSS_SYSTEM_GPS = BIT(0),
+	GNSS_SYSTEM_GLONASS = BIT(1),
+	GNSS_SYSTEM_GALILEO = BIT(2),
+	GNSS_SYSTEM_BEIDOU = BIT(3),
+	GNSS_SYSTEM_QZSS = BIT(4),
+	GNSS_SYSTEM_IRNSS = BIT(5),
+	GNSS_SYSTEM_SBAS = BIT(6),
+	GNSS_SYSTEM_IMES = BIT(7),
+};
```

Comment:
since you did a good job documenting all the enumerators, let's make sure these also get some love :)

```suggestion
enum gnss_system {
	/** Global Positioning System (GPS) */
	GNSS_SYSTEM_GPS = BIT(0),
	/** GLObal NAvigation Satellite System (GLONASS) */
	GNSS_SYSTEM_GLONASS = BIT(1),
	/** Galileo */
	GNSS_SYSTEM_GALILEO = BIT(2),
	/** BeiDou Navigation Satellite System */
	GNSS_SYSTEM_BEIDOU = BIT(3),
	/** Quasi-Zenith Satellite System (QZSS) */
	GNSS_SYSTEM_QZSS = BIT(4),
	/** Indian Regional Navigation Satellite System (IRNSS) */
	GNSS_SYSTEM_IRNSS = BIT(5),
	/** Satellite-Based Augmentation System (SBAS) */
	GNSS_SYSTEM_SBAS = BIT(6),
	/** Indoor Messaging System (IMES) */
	GNSS_SYSTEM_IMES = BIT(7),
};
```

Reply:
Committed your suggestion